ui/src/app/job-details/common/debug-icons/debug-icons.component.ts

```diff
@@ -69,22 +80,42 @@ export class JobDebugIconsComponent implements OnInit {
     return Object.keys(this.logFileData).includes(fileName) && this.logFileData[fileName] != '';
   }
 
+  hasOperationalDetails(): boolean {
+    return this.capabilities.authentication && this.capabilities.authentication.outsideAuth && !!this.operationId;
+  }
+
   showOrLinkTo(e: MouseEvent, url: string): void {
     e.stopPropagation();
     if (this.hasContents(this.getFileName(url))) {
-      this.logContentsDialog.open(JobLogContentsComponent, {
+      this.resourceContentsDialog.open(JobResourceContentsComponent, {
         disableClose: false,
         data: {
-          logName: this.getFileTitle(url),
-          logContents: this.logFileData[this.getFileName(url)],
-          logLink: this.getResourceUrl(url)
+          resourceName: this.getFileTitle(url),
+          resourceContents: this.logFileData[this.getFileName(url)],
+          resourceLink: this.getResourceUrl(url),
+          resourceType: 'text'
         }
       });
     } else if (url) {
       window.open(this.getResourceUrl(url));
     }
   }
 
+  showOperationDetails(): void {
+    this.jobManagerService.getOperationDetails(this.jobId, this.operationId)
+      .then((response) => {
+        this.resourceContentsDialog.open(JobResourceContentsComponent, {
+          disableClose: false,
+          data: {
+            resourceName: this.operationId,
+            resourceContents: new JsonPipe().transform(JSON.parse(response.details)),
```

Comment:
I'm noticing that in some places, like `resource-contents.component.ts`, we seem to check that the JSON exists before parsing, but not here.

Is there a danger that passing an empty string to the JSON parser will throw an exception that totally blows up the control flow?

Reply:
It does blow up with an empty string, so I've added a check – thanks!